src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy

```diff
@@ -466,7 +466,95 @@ class FlowStatSpec extends HealthCheckSpecification {
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+    }
+
+    @Tidy
+    def "System is able to collect stats after partial updating(vlan) on a flow endpoint"() {
+        given: "Two active neighboring switches connected to the traffgens"
+        def traffGenSwitches = topology.activeTraffGens*.switchConnected*.dpId
+        def switchPair = topologyHelper.switchPairs.find {
+            [it.src, it.dst].every { it.dpId in traffGenSwitches }
+        } ?: assumeTrue(false, "No suiting switches found")
+
+        and: "A flow with updated vlan on src endpoint via partial update"
+        def traffgenPortOnSrcSw = topology.activeTraffGens.find { it.switchConnected ==  switchPair.src}.switchPort
+
+        def flow = flowHelperV2.randomFlow(switchPair).tap { it.source.portNumber = traffgenPortOnSrcSw;  it.source.vlanId = 100}
+        flowHelperV2.addFlow(flow)
+
+        flowHelperV2.partialUpdate(flow.flowId, new FlowPatchV2().tap {
+            source = new FlowPatchEndpoint().tap { vlanId = vlanId ?: 100 + 1 }
+        })
+
+        when: "Generate traffic on the flow"
+        Date startTime = new Date()
+        def traffExam = traffExamProvider.get()
+        def exam = new FlowTrafficExamBuilder(topology, traffExam).buildExam(northbound.getFlow(flow.flowId),
+                (int) flow.maximumBandwidth, 5)
+        exam.setResources(traffExam.startExam(exam))
+        assert traffExam.waitExam(exam).hasTraffic()
+
+        then: "System collects stats for ingress/egress cookies"
+        def flowInfo = database.getFlow(flow.flowId)
+        def tags = [switchid: switchPair.src.dpId.toOtsdFormat(), flowid: flow.flowId]
+        def metric = metricPrefix + "flow.raw.bytes"
+        def waitInterval = 5
+        Wrappers.wait(statsRouterInterval, waitInterval) {
+            def forwardStats = otsdb.query(startTime, metric, tags + [cookie: flowInfo.forwardPath.cookie.value]).dps
+            assert forwardStats.size() > 0
+            assert forwardStats.values().each { it != 0 }
+            def reverseStats = otsdb.query(startTime, metric, tags + [cookie: flowInfo.reversePath.cookie.value]).dps
+            assert reverseStats.size() > 0
+            assert reverseStats.values().each { it != 0 }
+        }
+
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+    }
+
+    @Tidy
+    def "System is able to collect stats after partial updating(inner vlan) on a flow endpoint"() {
+        given: "Two active neighboring switches connected to the traffgens"
+        def traffGenSwitches = topology.activeTraffGens*.switchConnected*.dpId
+        def switchPair = topologyHelper.switchPairs.find {
+            [it.src, it.dst].every { it.dpId in traffGenSwitches }
+        } ?: assumeTrue(false, "No suiting switches found")
+
+        and: "A flow with updated inner vlan on src endpoint via partial update"
+        def traffgenPortOnSrcSw = topology.activeTraffGens.find { it.switchConnected ==  switchPair.src}.switchPort
```

Comment:
it is needless, you can create a flow on a traffgen port using the `randomFlow` method.
By default `randomFlow` tries to create a flow on the mentioned port.
```
  FlowRequestV2 randomFlow(SwitchPair switchPair, boolean useTraffgenPorts = true,
            List<FlowRequestV2> existingFlows = []) 
```
If you want explicitly set it, just `flowHelperV2.randomFlow(switchPair, true).tap {  it.source.vlanId = 100}`